web/tests/typeahead.test.js

```diff
@@ -150,6 +150,24 @@ run_test("triage", () => {
     );
 });
 
+run_test("triage: prioritise word boundary matches to arbitrary substring matches", () => {
+    const book = {name: "book"};
+    const hyphen_ok = {name: "hyphen_ok"};
+    const space_ok = {name: "space ok"};
```

Comment:
Let's include both `invalidnotok` and `invalidwith7ok` options that shouldn't match with a thing we're unlikely to consider a separator.